File: base/src/test/java/com/mindera/skeletoid/generic/UIUtilsUnitTests.java
package com.mindera.skeletoid.generic;

import android.content.Context;
import android.content.res.Resources;

import org.junit.Test;

import static junit.framework.Assert.assertEquals;
import static org.mockito.Matchers.any;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

public class UIUtilsUnitTests {

    @Test(expected = UnsupportedOperationException.class)
    public void testConstructor() {
        new UIUtils();
    }

    @Test
    public void testsGetStatusBarHeightInvalid() {
        Context context = mock(Context.class);
        Resources resources = mock(Resources.class);
        int invalidResourceId = -1;
        when(context.getResources()).thenReturn(resources);
        when(resources.getIdentifier(any(String.class), any(String.class), any(String.class))).thenReturn(invalidResourceId);

        assertEquals(0, UIUtils.getStatusBarHeighPixels(context));
    }

    @Test
    public void testsGetStatusBarHeight() {
        Context context = mock(Context.class);
        Resources resources = mock(Resources.class);
        int resourceId = 22;
        int statusBarHeight = 180;
        when(resources.getDimensionPixelSize(resourceId)).thenReturn(statusBarHeight);
        when(context.getResources()).thenReturn(resources);
        when(resources.getIdentifier(any(String.class), any(String.class), any(String.class))).thenReturn(resourceId);

        assertEquals(statusBarHeight, UIUtils.getStatusBarHeighPixels(context));
    }
}

File: base/src/test/java/com/mindera/skeletoid/generic/UIUtilsUnitTests.kt
package com.mindera.skeletoid.generic

import android.content.Context
import android.content.res.Resources
import com.mindera.skeletoid.generic.UIUtils.getStatusBarHeightPixels
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import org.mockito.ArgumentMatchers
import org.mockito.Mock
import org.mockito.Mockito
import org.mockito.MockitoAnnotations

class UIUtilsUnitTests {

    @Mock
    private lateinit var context: Context

    @Mock
    private lateinit var resources: Resources

    @Before
    fun setup() {
        MockitoAnnotations.initMocks(this)
    }

    @Test
    fun testsGetStatusBarHeightInvalid() {
        val invalidResourceId = -1

        Mockito.`when`(context.resources).thenReturn(resources)
        Mockito.`when`(
            resources.getIdentifier(
                ArgumentMatchers.any(String::class.java),
                ArgumentMatchers.any(String::class.java),
                ArgumentMatchers.any(String::class.java)
            )
        ).thenReturn(invalidResourceId)

        Assert.assertEquals(0, getStatusBarHeightPixels(context))
    }

    @Test
    fun testsGetStatusBarHeight() {
        val resourceId = 22
        val statusBarHeight = 180

        Mockito.`when`(resources.getDimensionPixelSize(resourceId)).thenReturn(statusBarHeight)
        Mockito.`when`(context.resources).thenReturn(resources)
        Mockito.`when`(
            resources.getIdentifier(
                ArgumentMatchers.any(String::class.java),
                ArgumentMatchers.any(String::class.java),
                ArgumentMatchers.any(String::class.java)
            )
        ).thenReturn(resourceId)

        Assert.assertEquals(statusBarHeight, getStatusBarHeightPixels(context))

    }
}